### PR TITLE
Dockerfile: use alpine to reduce size

### PR DESCRIPTION
**Pull request by tsao-chi**
Add compact, short information about your PR for easier understanding:

- Goal of the PR:   reduce size

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,32 +1,22 @@
...
+  git clone --depth=1 https://github.com/minetest/minetest_game.git /usr/src/minetest/games/minetest_game && \
+  rm -fr /usr/src/minetest/games/minetest_game/.git && \
+  cd /usr/src/minetest && \
+  cmake . \
+    -DCMAKE_INSTALL_PREFIX=/usr/local \
+    -DCMAKE_BUILD_TYPE=Release \
+    -DBUILD_SERVER=TRUE \
+    -DBUILD_CLIENT=FALSE && \
+  make -j2 && \
+  make install
```
Comment (SmallJoker) [inline Dockerfile]: Please use tabulators to indent. The code style guidelines may only mention C(++) and Lua, but other code should optimally follow similar rules.

File: Dockerfile [hunk 2]
```diff
@@ -1,32 +1,22 @@
...
-FROM debian:stretch
-
-USER root
-RUN groupadd minetest && useradd -m -g minetest -d /var/lib/minetest minetest && \
-    apt-get update -y && \
-    apt-get -y install libcurl3-gnutls libjsoncpp1 liblua5.1-0 libluajit-5.1-2 libpq5 libsqlite3-0 \
-        libstdc++6 zlib1g libc6 && \
-    apt-get clean && rm -rf /var/cache/apt/archives/* && \
-    rm -rf /var/lib/apt/lists/*
+RUN apk add --no-cache git build-base irrlicht-dev cmake bzip2-dev libpng-dev jpeg-dev libxxf86vm-dev mesa-dev sqlite-dev libogg-dev libvorbis-dev openal-soft-dev curl-dev freetype-dev zlib-dev gmp-dev jsoncpp-dev && \
```
Comment (SmallJoker) [inline Dockerfile]: Line is a bit long (>80 characters). Please break it for readability.

Comment (tsao-chi) [reply Dockerfile]: resolved

File: Dockerfile (hunk 1 above)
Comment (tsao-chi) [reply Dockerfile]: resolved

**Comment by rubenwardy** [comment]
by switching from debian to alphine, won't this break downstream users?

**Comment by tsao-chi** [comment]
@rubenwardy It may break mods with binaries or non-standard C code that depends on GCC

**Comment by SmallJoker** [comment]
tbh I don't see any benefit by switching the distribution, especially now that the build script works again.

**Comment by tsao-chi** [comment]
@SmallJoker 
reduce size.
The entire minetestserver image is even smaller than the `debian:stretch` image

**Comment by nerzhul** [comment]
can you rebase it please ?

**Comment by tsao-chi** [comment]
@nerzhul rebased

**Comment by nerzhul** [comment]
i merged #9643 changes into your PR

**Comment by nerzhul** [comment]
i will amend again today to add postgresql support

**Comment by tsao-chi** [comment]
@nerzhul no need to reduce it, it's build-time junk.

**Comment by nerzhul** [comment]
yep but for my usecase i have RUN_IN_PLACE in my workspace then i copy shit to the container and the build failed because it's already here.
I prefer a clean workspace, regarding or not it's a container :)

**Comment by nerzhul** [comment]
The MR is now ready to be merged

**Comment by nerzhul** [comment]
You have my approval for the switch to alpine, but we need a second approval :)

File: Dockerfile [hunk 3]
```diff
@@ -1,40 +1,52 @@
...
+COPY minetest.conf.example /usr/src/minetest/minetest.conf.example
+COPY builtin /usr/src/minetest/builtin
+COPY cmake /usr/src/minetest/cmake
+COPY doc /usr/src/minetest/doc
+COPY fonts /usr/src/minetest/fonts
+COPY lib /usr/src/minetest/lib
+COPY misc /usr/src/minetest/misc
+COPY po /usr/src/minetest/po
+COPY src /usr/src/minetest/src
+COPY textures /usr/src/minetest/textures
```
Comment (sfan5) [inline Dockerfile:17]: all these `COPY` instructions look prone to breakage, is `COPY . /usr/src/minetest` not possible?

Comment (nerzhul) [reply Dockerfile:17]: it's possible but it copies folders we don't want when local build is in RUN_IN_PLACE and the android build files, the build folders from cmake etc, it's junk for the build.
The main problem is the games folder which generally contains a minetest_game (like in mine) and here break the build on the git clone

**Comment by nerzhul** [comment]
You can find my post build output here: https://pastebin.com/kbqdNYYx

File: Dockerfile (hunk 3 above)
Comment (sfan5) [reply Dockerfile:17]: `/usr/src/minetest` isn't left in the final image, is it?
And for minetest_game it should be possible not to clone it, if it already exists.

**Pull request merged**